citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/AsyncCitationProcessor.java

```diff
@@ -100,6 +102,11 @@ public void sync(IUser user, String groupId, long contentVersion, String collect
         job.setStatus(JobStatus.PREPARED);
         jobRepo.save(job);
         jobManager.addJob(job);
+        Thread currentThread = Thread.currentThread();
```

Comment:
I think you can just do this in `checkIfThreadIsInterrupted` and remove the first parameter of the method.